File: 백준/Silver/1193. 분수찾기/분수찾기.py
x = int(input())
i=0
n=0
while n<x :
    i+=1
    n+=i
n-=i
a=x-n
b=i-x+n+1
if i%2 == 0 :
    print(str(a)+'/'+str(b))
else :
    print(str(b)+'/'+str(a))

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner scan = new Scanner(System.in);
        int input = scan.nextInt();
        int cnt = 1;
        int sum = 0;
        while(true){
            sum += cnt;
            if(sum>=input)
                break;
            cnt++;
        }
        int remain = input - (sum-cnt);
        if(cnt%2==1)
            System.out.println((cnt+1-remain)+"/"+remain);
        else
            System.out.println(remain+"/"+(cnt+1-remain));
    }
}